src/OrchardCore.Modules/OrchardCore.BackgroundTasks/Controllers/BackgroundTaskController.cs

```diff
@@ -270,12 +270,18 @@ private static BackgroundTaskSettings GetSettings(IBackgroundTask task, Backgrou
         {
             var name = task.GetTaskName();
 
-            if (document.Settings.ContainsKey(name))
+            BackgroundTaskSettings settings;
+            var defaultSettings = task.GetDefaultSettings();
+
+            if (document.Settings.TryGetValue(name, out settings))
             {
-                return document.Settings[name];
+                // Set the name using the default settings to ensure it populates old documents.
+                settings.Name = defaultSettings.Name;
```

Comment:
Here it depends if the `BackgroundTaskDocument document` was retrieved by using `LoadDocumentAsync()` or by using `GetDocumentAsync()`, the first returns a mutable document that we use in the Post Edit, but the second returns an immutable document that should not be updated. So here we would need to make the distinction.


    /// <summary>
    /// Loads the background task document from the store for updating and that should not be cached.
    /// </summary>
    public Task<BackgroundTaskDocument> LoadDocumentAsync() => _documentManager.GetOrCreateMutableAsync();

    /// <summary>
    /// Gets the background task document from the cache for sharing and that should not be updated.
    /// </summary>
    public Task<BackgroundTaskDocument> GetDocumentAsync() => _documentManager.GetOrCreateImmutableAsync();

Reply:
@jtkech yes but it should not matter in this case. Both get requests "List and Edit" are being set just for UI purposes. The records are being read and not saved. The post request to the edit method is where we actually update the database and there we already use Load. This change will fix it wether we want to save or just show it.

Reply:
Even if here we only update an immutable string, we should not mutate a shared cached object.

Reply:
Are you saying we should use Load method everywhere in this case?

Reply:
No, better to still use the Get method in the Get Edit.

Then would need to look at the code, but normally only set the view model with the desired name.

Reply:
I'll look at the code more tomorrow

Reply:
@jtkech I made some changes to the code.

Reply:
Okay, looks good!